occurrence-download/src/main/java/org/gbif/occurrence/download/file/specieslist/SpeciesListCollector.java

```diff
@@ -0,0 +1,42 @@
+package org.gbif.occurrence.download.file.specieslist;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.gbif.dwc.terms.GbifTerm;
+import org.gbif.occurrence.download.hive.DownloadTerms;
+
+public class SpeciesListCollector {
+  
+  private final List<Map<String,String>> distinctSpeciesRecord = new ArrayList<>();
+  /**
+   * @return list of records of distinct species.
+   */
+  public List<Map<String, String>> getDistinctSpecies() {
+    return distinctSpeciesRecord;
+  }
+  
+  /**
+   * group results by taxon key and order them in {@link DownloadTerms} species list download order.
+   * @return distinct species
+   */
+  public SpeciesListCollector computeDistinctSpecies(List<Map<String,String>> filteredResult) {
```

Comment:
there's no point in returning "this" here since this is not a chainable method, better if aggregate results as we are collecting them